crates/spirv-builder/src/test/basic.rs

```diff
@@ -126,3 +126,52 @@ pub fn main() {
 }
 "#);
 }
+
+// NOTE(eddyb) this won't pass Vulkan validation (see `push_constant_vulkan`),
+// but should still pass the basline SPIR-V validation.
+#[test]
+fn push_constant() {
+    val(r#"
+#[derive(Copy, Clone)]
+pub struct ShaderConstants {
+    pub width: u32,
+    pub height: u32,
+    pub time: f32,
+}
+
+#[allow(unused_attributes)]
+#[spirv(fragment)]
+pub fn main(
+    #[spirv(push_constant)] constants: PushConstant<ShaderConstants>,
```

Comment:
I believe `#[spirv(push_constant)]` does nothing and was erroneously included in the example shader, which I'm assuming you copied this from. Doesn't really matter though.

Reply:
Oh haha whoops.